src/server/pkg/worker/api_server.go

```diff
@@ -269,6 +274,15 @@ func (a *APIServer) uploadOutput(ctx context.Context, tag string) error {
 				return nil
 			}
 
+			// Under some circumstances, the user might have copied
+			// some pipes from the input directory to the output directory.
+			// Reading from these files will result in job blocking.  Thus
+			// we preemptively detect if the file is a special file.
+			if (info.Mode() | os.ModeType) > 0 {
```

Comment:
Just to check: does this work for directories? I took a quick look at the docs for ModeType, and it looked like
```
  ModeType = ModeDir | ModeSymlink | ModeNamedPipe | ModeSocket | ModeDevice
```
so I just want to check that users won't get 'cannot upload special file' just because the output contains a directory